src/org/daisy/dotify/formatter/impl/core/TableOfContentsImpl.java

```diff
@@ -97,30 +133,118 @@ public void startEntry(String refId) {
 	
 	@Override
 	public void endEntry() {
+        if (!inTocEntry) {
+            throw new RuntimeException("Unexpected end of toc-entry");
+        }
 		inTocEntry = false;
 	}
 
+    @Override
+    public FormatterCoreImpl getEntryOnResumed() {
+        assertInTocEntryOnResumed();
+        return tocEntryOnResumedBlocks;
+    }
+
+    @Override
+	public void startEntryOnResumed(String range) {
+		if (inTocEntry || inTocEntryOnResumed) {
+			throw new RuntimeException("toc-entry and toc-entry-on-resumed may not be nested");
+		}
+		inTocEntryOnResumed = true;
+        hasTocEntryOnResumed = true;
+        
+        String startRefId = null;
```

Comment:
Why not keep this code inside `TocEntryOnResumedRange`?

Reply:
Done.